com/course/zenkouv/src/main/java/lecture6/httpComponent/HttpComponentExample.java

```diff
@@ -0,0 +1,45 @@
+package lecture6.httpComponent;
+
+import lecture6.constant.GitHubApiConstants;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+public class HttpComponentExample {
+
+    private static final CloseableHttpClient httpClient = HttpClients.custom().build();
+
+    private HttpComponentExample() {
+    }
+
+    public static void sendCollaboratorInvite(String name) throws IOException, URISyntaxException {
```

Comment:
let's split  all this code to few levels
1) business level  
should contains methods like  sendCollaboratorInvite (String name) String getCollaborators() without technical low level details. Also it will be responsible from parsing responses to models 
2) controllers - responsible for combine endpoint, params, method (post put) etc .
3) technical level - create interface IRequestExecutor   with method doRequest(...) and encapsulate request logic there  

create few implementation for HttpClient and restAssured

Reply:
fix